include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,767 @@
+            {   // proceed within the view since we are right of the previous gap but didn't arrive at the right gap yet
+                ++ungapped_view_pos;
+            }
+            else
+            {   // we arrived at the right gap and have to update the variables. ungapped_view_pos remains unchanged.
+                left_gap_end = (*anchor_set_it).first + (*anchor_set_it).second -
+                               ((anchor_set_it != (*host).anchors.begin()) ?
+                                    (*std::prev(anchor_set_it)).second : 0);
+                ++anchor_set_it;
+            }
+
+            return *this;
+        }
+
+        //!\brief Pre-decrement, returns updated iterator.
+        constexpr gap_decorator_anchor_set_iterator & operator--() noexcept
+        {
+            assert((host)); // host is set
+            --pos;
+
+            if (pos < left_gap_end)
+            {   // there was no gap before but we arrive at the left gap and have to update the variables.
+                if (anchor_set_it != (*host).anchors.begin())
+                {
+                    --anchor_set_it;
+                    if (anchor_set_it != (*host).anchors.begin())
+                    {
+                        auto prev = std::prev(anchor_set_it);
+                        left_gap_end = (*prev).first + (*prev).second -
+                                       ((prev != (*host).anchors.begin()) ? (*std::prev(prev)).second : 0);
+                    }
+                    else // [[unlikely]]
+                    {
+                        left_gap_end = 0;
+                    }
+                }
+            }
+            else if (anchor_set_it == (*host).anchors.end() || pos < (*anchor_set_it).first)
+            {   // we are neither at the left nor right gap
+                --ungapped_view_pos;
+            }
+            // else -> no op (we are still within the right gap stretch)
+
+            return *this;
+        }
+
+        //!\brief Post-increment, returns previous iterator state (delegates to pre-increment).
+        constexpr gap_decorator_anchor_set_iterator operator++(int) noexcept
+        {
+            gap_decorator_anchor_set_iterator cpy{*this};
+            ++(*this);
+            return cpy;
+        }
+
+        //!\brief Post-decrement, returns previous iterator state (delegates to pre-decrement).
+        constexpr gap_decorator_anchor_set_iterator operator--(int) noexcept
+        {
+            gap_decorator_anchor_set_iterator cpy{*this};
+            --(*this);
+            return cpy;
+        }
+        //!\}
+
+        /*!\name Reference/Dereference operators
+         * \{
+        */
+        //!\brief Dereference operator returns element currently pointed at.
+        constexpr reference operator*() const noexcept
+        {
+            if (pos >= left_gap_end && (anchor_set_it == (*host).anchors.end() || pos < (*anchor_set_it).first))
```

Comment:
This has to be evaluated on every deref now? Could/should this be cached?

Reply:
I guess this can be cashed..

Reply:
when I cash it I need again more if branches in the increment/decrement